rust_src/src/buffers.rs

```diff
@@ -1982,4 +2030,268 @@ pub fn byte_char_debug_check(b: LispBufferRef, charpos: isize, bytepos: isize) {
+            return unbind_to(count, Qnil).into();
+        }
+    }
+
+    // If the hooks have killed the buffer, exit now.
+    if !b.is_live() {
+        return unbind_to(count, Qt).into();
+    }
+
+    // Then run the hooks.
+    unsafe { run_hook(Qkill_buffer_hook) };
+    unbind_to(count, Qnil);
+
+    // If the hooks have killed the buffer, exit now.
+    if !b.is_live() {
+        return true;
+    }
+
+    // We have no more questions to ask.  Verify that it is valid
+    // to kill the buffer.  This must be done after the questions
+    // since anything can happen within do_yes_or_no_p.
+
+    // Don't kill the minibuffer now current.
+    if LispObject::from(b) == LispWindowRef::from(unsafe { minibuf_window }).contents {
+        return false;
+    }
+
+    // When we kill an ordinary buffer which shares its buffer text
+    // with indirect buffer(s), we must kill indirect buffer(s) too.
+    // We do it at this stage so nothing terrible happens if they
+    // ask questions or their hooks get errors.
+    if b.base_buffer.is_null() && b.indirections > 0 {
+        for other in LispBufferIter::new() {
+            if other.base_buffer == b.as_mut() {
+                kill_buffer(Some(other.into()));
+            }
+        }
+
+        // Exit if we now have killed the base buffer (Bug#11665).
+        if !b.is_live() {
+            return true;
+        }
+    }
+
+    // Run replace_buffer_in_windows before making another buffer current
+    // since set-window-buffer-start-and-point will refuse to make another
+    // buffer current if the selected window does not show the current
+    // buffer (bug#10114).
+    unsafe { replace_buffer_in_windows(b.into()) };
+
+    // Exit if replacing the buffer in windows has killed our buffer.
+    if !b.is_live() {
+        return true;
+    }
+
+    // Make this buffer not be current.  Exit if it is the sole visible
+    // buffer.
+    if b == current_buffer().into() {
+        set_buffer(LispBufferRef::from(unsafe { Fother_buffer(b.into(), Qnil, Qnil) }).into());
+        if b == current_buffer().into() {
+            return false;
+        }
+    }
+
+    // If the buffer now current is shown in the minibuffer and our buffer
+    // is the sole other buffer give up.
+    if current_buffer() == LispWindowRef::from(unsafe { minibuf_window }).contents
+        && LispObject::from(b) == unsafe { Fother_buffer(b.into(), Qnil, Qnil) }
+    {
+        return false;
+    }
+
+    // Now there is no question: we can kill the buffer.
+
+    // Unlock this buffer's file, if it is locked.
+    unsafe {
+        unlock_buffer(b.as_mut());
+
+        kill_buffer_processes(b.into());
+        kill_buffer_xwidgets(b.into());
+    }
+
+    // Killing buffer processes may run sentinels which may have killed
+    // our buffer.
+    if !b.is_live() {
+        return true;
+    }
+
+    // These may run Lisp code and into infinite loops (if someone
```

Comment:
Is this our grammar issue or was it in the original code?

Reply:
That was in the original, most comments have been copied verbatim

Reply:
Let's drop the word 'and'.